utils/build-script

```diff
@@ -63,7 +62,718 @@ def call_without_sleeping(command, dry_run=False):
         # Don't mutate the caller's copy of the arguments.
         command = ["caffeinate"] + list(command)
 
-    shell.call(command, dry_run=dry_run, echo=False)
+    shell.call(command, env=env, dry_run=dry_run)
+
+
+class HostSpecificConfiguration(object):
+    """Configuration information for an individual host."""
+
+    def __init__(self, host_target, invocation):
+        """Initialize for the given `host_target`."""
+
+        # Compute the set of deployment targets to configure/build.
+        args = invocation.args
+        if host_target == args.host_target:
+            # This host is the user's desired product, so honor the requested
+            # set of targets to configure/build.
+            needed_deployment_targets = args.stdlib_deployment_targets
+            if "all" in args.build_stdlib_deployment_targets:
+                needed_deployment_targets_to_build = set(
+                    needed_deployment_targets)
+            else:
+                needed_deployment_targets_to_build = set(
+                    args.build_stdlib_deployment_targets).intersect(
+                        set(args.stdlib_deployment_targets))
+        else:
+            # Otherwise, this is a host we are building as part of
+            # cross-compiling, so we only need the target itself.
+            needed_deployment_targets = [host_target]
+            needed_deployment_targets_to_build = set(needed_deployment_targets)
+
+        # Compute the lists of **CMake** targets for each use case and the SDKs
+        # to configure with.
+        #
+        # NOTE: We frequently (when building on Darwin) configure a larger
+        # set of targets than we build for by default. This is so that
+        # developers can easily build for iOS without reconfiguring.
+        self.stdlib_targets_to_build = []
+        self.targets_to_test = []
+        self.benchmark_targets_to_build = []
+        self.benchmark_targets_to_run = []
+        self.sdks_to_configure = set()
```

Comment:
1. Can we reorder these, so the configure targets come first?

2. I don't really like the names here. It starts with `needed_deployment_targets` - how/why is it needed? It should be called `stdlib_targets_to_configure`, but that's a bit strange because it would look like `stdlib_targets_to_build`, except `_to_build` doesn't actually contain `stdlib-deployment-target`s, it contains Swift build targets.

I would suggest the following:
```
needed_deployment_targets -> stdlib_targets_to_configure
needed_deployment_targets_to_build -> stdlib_targets_to_build
stdlib_targets_to_build -> swift_stdlib_build_targets
targets_to_test -> swift_test_run_targets
benchmark_targets_to_build -> swift_benchmark_build_targets
benchmark_targets_to_run -> swift_benchmark_run_targets
```

These are all internal to the build script, so as long as we pass the correct ones along to `-impl` we should be okay.